src/main/java/org/assertj/core/internal/Maps.java

```diff
@@ -550,12 +551,37 @@ public void assertHasSameSizeAs(AssertionInfo info, Map<?, ?> map, Map<?, ?> oth
    */
   public <K, V> void assertContainsOnlyKeys(AssertionInfo info, Map<K, V> actual,
                                             @SuppressWarnings("unchecked") K... keys) {
+    assertContainsOnlyKeys(info, actual, "array of keys", keys);
+  }
+
+  /**
+   * Verifies that the actual map contains only the given keys and nothing else, in any order.
+   *
+   * @param <K> key type
+   * @param <V> value type
+   * @param info contains information about the assertion.
+   * @param actual the given {@code Map}.
+   * @param keys the keys that are expected to be in the given {@code Map}.
+   * @throws NullPointerException if the array of keys is {@code null}.
+   * @throws IllegalArgumentException if the array of keys is empty.
+   * @throws AssertionError if the given {@code Map} is {@code null}.
+   * @throws AssertionError if the given {@code Map} does not contain the given keys or if the given {@code Map}
+   *           contains keys that are not in the given array.
+   */
+  public <K, V> void assertContainsOnlyKeys(AssertionInfo info, Map<K, V> actual,
+                                            Iterable<K> keys) {
+    final K[] keysAsArray = keys != null ? (K[]) Streams.stream(keys).toArray(Object[]::new) : null;
+    assertContainsOnlyKeys(info, actual, "keys iterable", keysAsArray);
+  }
+
+  private <K, V> void assertContainsOnlyKeys(AssertionInfo info, Map<K, V> actual,
+                                             String placeholderForErrorMessages, K... keys) {
     assertNotNull(info, actual);
-    failIfNull(keys);
+    failIfNull(keys, ErrorMessages.keysToLookForIsNull(placeholderForErrorMessages));
     if (actual.isEmpty() && keys.length == 0) {
       return;
     }
-    failIfEmpty(keys);
+    failIfEmpty(keys, ErrorMessages.keysToLookForIsEmpty(placeholderForErrorMessages));
```

Comment:
static import `keysToLookForIsNull` for readability

Reply:
done